src/OT/glyf/glyf.hh

```diff
@@ -389,6 +403,30 @@ glyf::_populate_subset_glyphs (const hb_subset_plan_t   *plan,
   ;
 }
 
+inline void
+glyf::_compile_subset_glyphs_with_deltas (const hb_subset_plan_t *plan,
```

Comment:
This compilation allocates an individual buffer for each glyph. It's probably better to create a second serialization buffer the same size as the final serialization buffer and then write the compiled glyph byte blobs into the serialization buffer. Then the final serialization will transfer them into the final positions with appropriate padding in the final serialization context.

Reply:
Uh, I think maybe we don't need a second buffer cause top level glyf table serializes glyphs one by one. See:  https://github.com/harfbuzz/harfbuzz/blob/main/src/OT/glyf/glyf.hh#L50, and padding is handled during each SubsetGlyph serialization: https://github.com/harfbuzz/harfbuzz/blob/main/src/OT/glyf/SubsetGlyph.hh#L20

Reply:
The most important thing is it would be nice to use a serialize context throughout the compilation could instead of memcopying into raw buffers.

Reply:
Also agree.

Reply:
Here's my understanding for the original subset code: we need to know if `use_short_loca` is true before we could write each glyph into serializer buffer(padding depends on this) https://github.com/harfbuzz/harfbuzz/blob/main/src/OT/glyf/glyf.hh#L84. 
And in order to know if we use short loca, we need to compile each glyph first and then compute the sum of each glyph's (length+padding) size. Thus we can not directly use serializer at first.  Not sure if there's another better way to do this.

Reply:
Right, I was thinking you could create a buffer + a new serialize context in this function then pass that down to all of the encoding/compiling operations to write into. Each glyph could be packed as an object in the serialization resulting in an object id. Then with the object id's you could extract the head and tail positions for each glyph in the buffer.